### PR TITLE
Implement Kitti generator

### PR DESCRIPTION
**Pull request by lvaleriu**
I started a new pull request from scratch. Hopefully i made it easier for you to follow the code.

### CODE REVIEW

File: keras_retinanet/preprocessing/kitti.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    'Cyclist': 5,
+    'Tram': 6,
+    'Misc': 7,
+    'DontCare': 7
+}
+
+
+class KittiGenerator(Generator):
+    def __init__(
+            self,
```
Comment (hgaiser) [inline keras_retinanet/preprocessing/kitti.py]: Too much indentation.

File: keras_retinanet/preprocessing/kitti.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        self.image_data = dict()
+        self.images = []
+        for i, fn in enumerate(os.listdir(label_dir)):
+            label_fp = os.path.join(label_dir, fn)
+            image_fp = os.path.join(image_dir, fn.replace('.txt', '.png'))
+
+            self.images.append(image_fp)
+
+            with open(label_fp, 'r') as csv_file:
+                reader = csv.DictReader(csv_file, delimiter=' ',
```
Comment (hgaiser) [inline keras_retinanet/preprocessing/kitti.py]: Reformat as:
```python
foo(
    bar,
    baz=[
        1, 2, 3,
        4, 5, 6
    ]
)
```

Comment (lvaleriu) [reply keras_retinanet/preprocessing/kitti.py]: Done

File: keras_retinanet/preprocessing/kitti.py (hunk 1 above)
Comment (lvaleriu) [reply keras_retinanet/preprocessing/kitti.py]: Done!

**Comment by awilliamson** [comment]
@hgaiser This could cause issues with #284, same ones I had when trying to work around the OID introduction. Couldn't really rebase nicely, had to make a merge commit. It would then allow the wrapping of the kitti preprocessing script to be entirely self-contained within the KITTI plugin module, if @lvaleriu wished to go that route.

**Comment by lvaleriu** [comment]
Go for the plugin module!

**Comment by de-vri-es** [comment]
Dataset plugins won't have to block merging this. I can rebase the branch after.

File: keras_retinanet/preprocessing/kitti.py (hunk 2 above)
Comment (hgaiser) [reply keras_retinanet/preprocessing/kitti.py]: Could you reformat this still?

File: keras_retinanet/preprocessing/kitti.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                                            'dh', 'dw', 'dl',
+                                            'lx', 'ly', 'lz', 'ry'
+                                        ])
+                boxes = []
+                for line, row in enumerate(reader):
+                    label = row['type']
+                    cls_id = kitti_classes[label]
+
+                    annotation = {'cls_id': cls_id, 'x1': row['left'], 'x2': row['right'],
+                                  'y2': row['bottom'], 'y1': row['top']}
```
Comment (hgaiser) [inline keras_retinanet/preprocessing/kitti.py]: And this too, similarly as the other one.

**Comment by hgaiser** [comment]
@lvaleriu can you update this? I'd like to merge it.

**Comment by lvaleriu** [comment]
@hgaiser For some reasons I didnt receive any notifications for your comments. Strange .... (or I'm blind).
Ok, i'll take a look right away! :)

**Comment by lvaleriu** [comment]
@hgaiser  Tell me if everything is ok now.

**Comment by hgaiser** [comment]
Thanks :+1:

**Pull request merged**